library_samples/Python3/ocs_sample_library_preview/DataViews.py

```diff
@@ -267,58 +286,64 @@ def getDataInterpolated(
         params = {
             "count": count,
             "form": form,
-            "continuationToken": continuationToken,
             "startIndex": startIndex,
             "endIndex": endIndex,
-            "interval": interval,
+            "interval": interval
         }
-        response = requests.get(
-            self.__getDataInterpolated.format(
-                tenant_id=self.__baseClient.tenant,
-                namespace_id=namespace_id,
-                dataView_id=dataView_id,
-            ),
-            headers=self.__baseClient.sdsHeaders(),
-            params=params,
-        )
+        response = {}
+        if url:
+            response = requests.get(url)
+        else:
+            response = requests.get(
+                self.__dataViewDataInterpolated.format(
+                    tenant_id=self.__baseClient.tenant,
+                    namespace_id=namespace_id,
+                    dataView_id=dataView_id,
+                ),
+                headers=self.__baseClient.sdsHeaders(),
+                params=params,
+            )
 
         self.__baseClient.checkResponse(
             response,
             f"Failed to get DataView data interpolated for DataView, {dataView_id}.",
         )
 
-        continuation_token = None
-        next_page = response.headers.get("NextPage", None)
-        if next_page:
-            token_param = "&continuationToken="
-            token_position = next_page.find(token_param)
-            assert token_position > 0, "Could not find continuationToken in NextPage"
-            end_position = next_page.find("&", token_position+1)
-            end_position = None if end_position == -1 else end_position
-            continuation_token = next_page[token_position +
-                                           len(token_param):end_position]
+        nextPage = None
+        firstPage = None
+
+        if hasattr(response.headers, "NextPage"):
+            nextPage = response.headers["NextPage"]
+
+        if hasattr(response.headers, "FirstPage"):
+            firstPage = response.headers["FirstPage"]
 
         if form is not None:
-            return response.text, continuation_token
+            return response.text, nextPage, firstPage
 
         content = response.json()
 
         if value_class is None:
-            return content, continuation_token
-        return value_class.fromJson(content), continuation_token
+            return content, nextPage, firstPage
+        return value_class.fromJson(content), nextPage, firstPage
 
     def __setPathAndQueryTemplates(self):
         """
         Internal  Sets the needed URLs
         :return:
         """
         self.__basePath = (
-            self.__baseClient.uri_API +
+            self.__baseClient.uri_API + "-preview"
```

Comment:
Are we sure we want "preview" hardcoded?

Reply:
i guess i was figuring we might have updates when it goes to not preview.  right now it shares the same base client as the SDS libs, which means if we bring the v1-preview from the config we would need a special entry for data views and reference it there... or the sds part of the sample won't work since it does work in the preview route

Reply:
Understood, this is probably the best case then.